the-collab-lab/tcl-76-smart-shopping-list
# Mp create snap shop account

## Description

As a user, I want an account connected to SnapShop


This code allows for a user to create an account using a google email. Once they have created an account, they will be redirected to the homepage to start creating lists. The Create Account link is conditionally rendered dependent on if someone is signed in or not. 


## Acceptance Criteria

- [ ] There is a Create an Account feature where users can input a username and password
- [ ] The username and password created by the user is added to our Firebase dataset
- [ ] The user can log in and out using the username and password combination they created
- [ ] Users lists are connected to their username and password, and users cannot access other accounts' lists


## Type of Changes

`feature`

## Testing Steps / QA Criteria
1. Create a new google email account
2. Go sign up using the create account page on SnapShop with the google email you just created and a password
3. If the creation is successful, you will get a success message and be redirected to the homepage
4. Check firebase console to see your new account linked
5. Sign out of the app, then try signing back in and check if items added and lists are updated for the correct user
